include/behaviortree_cpp_v3/bt_factory.h

```diff
@@ -240,9 +240,13 @@ class Tree
 
   void updateGeneralStatus()
   {
-    if (rootNode() && rootNode()->getGeneralStatus().has_value())
+    if (rootNode())
     {
-      general_status = buildTreeGeneralStatus(rootNode());
+      if (rootNode()->getGeneralStatus().has_value())
+      {
+        general_status = buildTreeGeneralStatus(rootNode());
+      }
+      resetTreeGeneralStatus(rootNode());
```

Comment:
Is it ok to reset the general status when the root node doesn't have a general status? Are we sure that all the nodes have a general status at the same time?

Is the following situation possible?

rootNode()->getGeneralStatus().has_value() -> false
childNode()->getGeneralStatus().has_value() -> true

Reply:
Yes, we can reset any node regardless of the state of generalStatus. 
The tree should reset the statuses by itself when it is halted, this `resetTreeGeneralStatus` is just a precaution to cleanup the tree. I added it after reading comments in `haltTree()`:

```
    // the halt should propagate to all the node if the nodes
    // have been implemented correctly
    rootNode()->halt();
    rootNode()->setStatus(NodeStatus::IDLE);

    //but, just in case.... this should be no-op
    auto visitor = [](BT::TreeNode* node) {
      node->halt();
      node->setStatus(BT::NodeStatus::IDLE);
    };
    BT::applyRecursiveVisitor(rootNode(), visitor);
```